engine/src/main/java/org/terasology/rendering/dag/DefaultStateChangeStorage.java

```diff
@@ -15,20 +15,19 @@
  */
 package org.terasology.rendering.dag;
 
-import java.util.List;
+import java.util.HashMap;
+import static org.lwjgl.opengl.GL11.GL_FILL;
 
 /**
- *
+ * TODO: Add javadocs
  */
-public class RenderPipelineExecutor {
-
-    private List<PipelineTask> taskList;
+class DefaultStateChangeStorage extends HashMap<Class<? extends StateChange>, StateChange> {
```

Comment:
And if you really want to pick up a mini-challenge you could make sure that 

SetColorStateChange("pink") == SetColorStateChange("pink") is always true, 

that is, for each possible value of a state change a singleton is generated, so that both the intermediate pipeline list and the final render task list may contain identical items that are ultimately the exact same object, reused multiple times. Make sense?

It's not really required though. It's just a nice thing to have. And it would be most useful in the RenderTaskList.